### ISSUE TITLE
The triton operators performed worse after fusion

### ISSUE THREAD

**Issue by qingshanxwx** [pytorch/pytorch#135994]
### 📚 The doc issue

Platforms:linux
torch version: 2.1
triton version: 2.1

![image (2)](https://github.com/user-attachments/assets/bde916ee-a62d-4c78-87ee-4a9ffdd04e4f)

On our hardware, when the torch.compile use max-autotune to pprocess this section of subgraph, The addmm will select the
triton_mm operator during tuning, and then it fused with the gelu and dropout into a single triton_kernel.
The triton_kernel send a triton_unk_fused_addmm_gelu_native_dropout to GPU. the kernel takes 450us on GPU.
![image](https://github.com/user-attachments/assets/7fb61404-f585-4426-af2a-87194cecef9a)
After that, we force addmm not to participate in the fusion. The adcdmm will send a gemm operator to GPU, the gelu and dropout will fuse and send a triton_poi_fused_gelu_native_dropout to GPU. 1The GPU total takes 250us

We did the same experiment on NV's A800:
On the A800, The addmm will select the bias_addmm operator during tuning, the gelu and dropout fused into a
triton_poi_fused_gelu_native_dropout, the kernel takes100us on GPU
![image (3)](https://github.com/user-attachments/assets/82e27826-bdf6-4f5c-8dbd-ffedf4d7edb4)

Finally, on the A800 we forced addmm to select the triton_mm operator during tuning and fused with the gelu, dropout, it would
also performed worse and it take total 300us on GPU.

Based on the above, we have a question:
Should there be a fusion strategy to solve this problem?

The test code as follows
```
import torch.nn as nn
import torch
from torch.profiler import profile, ProfilerActivity

class PositionwiseFeedForward(nn.Module):
    "Implements FFN equation."

    def __init__(self, d_model=768, d_ff=3072, dropout=0.1):
        super(PositionwiseFeedForward, self).__init__()
        self.w_1 = nn.Linear(d_model, d_ff, dtype=torch.float16)
        self.dropout = nn.Dropout(dropout)
        self.activation = nn.GELU()

    def forward(self, x):
        return self.dropout(self.activation(self.w_1(x)))

x = torch.rand((16,128,768), dtype=torch.float16).cuda()

op = PositionwiseFeedForward().cuda()

for i in range(30):
    if i ==15:
        with profile(activities=[ProfilerActivity.CPU, ProfilerActivity.CUDA], record_shapes=True) as prof1:
            op(x)
        continue
    op(x)

print(prof1.key_averages(group_by_input_shape=True).table(sort_by='self_cpu_time_total'))
prof1.export_chrome_trace('eager.json')

    
c_op = torch.compile(op, mode='max-autotune')

for i in range(30):
    if i ==15:
        with profile(activities=[ProfilerActivity.CPU, ProfilerActivity.CUDA], record_shapes=True) as prof2:
            c_op(x)
        continue
    c_op(x)

print(prof2.key_averages(group_by_input_shape=True).table(sort_by='self_cuda_time_total'))
prof2.export_chrome_trace('compile.json')
```

**Comment by eellison** [comment]
Hi, this issue is fixing on master. See [benchmark_epilogue_fusion](
https://github.com/pytorch/pytorch/blob/b6d6aa49b839b11c6bb01688975d2ef77c3936d0/torch/_inductor/config.py#L428-L430)
